src/zcash/prf.h

```diff
@@ -0,0 +1,24 @@
+/*
+Zcash uses SHA256Compress as a PRF for various components
+within the zkSNARK circuit.
+*/
+
+#ifndef _PRF_H_
+#define _PRF_H_
+
+#include "uint256.h"
+
+uint256 PRF(bool a, bool b, bool c, bool d,
+            uint256 x,
+            uint256 y);
+
+template<unsigned char T>
+uint256 PRF_addr(uint256 a_sk)
+{
+    uint256 y;
+    *(y.begin()) = T;
```

Comment:
`uint256` does not have endianness properties; it is not a number, despite the misleading name. (We have to convert it into a `arith_uint256` in order to perform arithmetic on it, for example.) It is a pure hash digest, a transparent blob of 32 bytes, and can be freely converted to and from a `vector<unsigned char>`.